File: Arrays and Hashing/705-Design-HashSet/705.js

var MyHashSet = function() {
    this.setArray = new Array(10000);
    for(let i = 0; i < 10000; i++){
        this.setArray[i] = [];
    }
};

/** 
 * @param {number} key
 * @return {void}
 */
MyHashSet.prototype.add = function(key) {
    const idx = key % 10000;
    if (!this.setArray[idx].includes(key)) {
        this.setArray[idx].push(key);
    }
};

/** 
 * @param {number} key
 * @return {void}
 */
MyHashSet.prototype.remove = function(key) {
    const idx = key % 10000;
    const index = this.setArray[idx].indexOf(key);
    if (index !== -1) {
        this.setArray[idx].splice(index, 1);
    }
};

/** 
 * @param {number} key
 * @return {boolean}
 */
MyHashSet.prototype.contains = function(key) {
    const idx = key % 10000;
    return this.setArray[idx].includes(key);
};

/** 
 * Your MyHashSet object will be instantiated and called as such:
 * var obj = new MyHashSet()
 * obj.add(key)
 * obj.remove(key)
 * var param_3 = obj.contains(key)
 */

File: Arrays and Hashing/705-Design-HashSet/705.cs
public class MyHashSet {
    
    private List<int>[] setArray;

    public MyHashSet() {
        setArray = new List<int>[10000];
        for (int i = 0; i < 10000; i++) {
            setArray[i] = new List<int>();
        }
    }
    
    public void Add(int key) {
        int idx = key % 10000;
        if (!setArray[idx].Contains(key)) {
            setArray[idx].Add(key);
        }
    }
    
    public void Remove(int key) {
        int idx = key % 10000;
        setArray[idx].Remove(key);
    }
    
    public bool Contains(int key) {
        int idx = key % 10000;
        return setArray[idx].Contains(key);
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.Add(key);
 * obj.Remove(key);
 * bool param_3 = obj.Contains(key);
 */